### PR TITLE
Initial submission for review

### PR DESCRIPTION
**Pull request by gmkv**
This pull request introduces code for the exercises on the second chapter of the Go course.
- The small package **mathfunc** provides functionality to calculate GCD and LCM of two numbers in order to demonstrate how basic language constructs work in Go.
- **arrayutil** is a set of documented functions that work over arrays of integers. It includes functionality to gather the minimal element of the array, to get the sum of all numbers, to sort the array, to reverse it, and to print the array to the console.
- **arbstring** provides a way to generate a string of random alphanumeric characters of arbitrary lengths.
- Package **visibilitydemo** demonstrates how Golang realises visibility or methods, variables, constants and types. The usage examples are written in `exp-unexp.go `

### CODE REVIEW

File: arbstring/arbstring.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+package arbstring
+
+import (
+	"math/rand"
+	"time"
+)
+
+// Generate returns a slice of strings with length, each element, filled with random characters
+func Generate(length int) string {
+	var symbols = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
```
Comment (mgenov) [inline arbstring/arbstring.go]: you can defined it as const and move it out of the function.

File: arrayutil/arrayutil.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	return result
+}
+
+// Sum returns the total sum of the numbers in the array
+func Sum(array []int) int {
+	if emptyb(array) {
+		return 0
+	}
+	var result int
+	for i := 0; i < len(array); i++ {
```
Comment (mgenov) [inline arrayutil/arrayutil.go]: You can simplify it with range loop.

**Comment by mgenov** [comment]
The description of the PR should mention details for the change.

**Comment by gmkv** [comment]
I have made the two requested changes. Can you let me know is appropriate to write in the description of the PR? Should I go into more detail into how each exercise has been solved?

**Comment by mgenov** [comment]
Not in complete details, but something like: "Added an array utility package for dealing with arrays whose goal is to show the basic language usages of the Go language"

File: arrayutil/arrayutil.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	}
+	return result
+}
+
+// Sum returns the total sum of the numbers in the array
+func Sum(array []int) int {
+	if emptyb(array) {
+		return 0
+	}
+	var result int
```
Comment (mgenov) [inline arrayutil/arrayutil.go]: result := 0

**Comment by gmkv** [comment]
I have made the change in arrayutil.go and have rewritten the pull request description to better explain the proposed pull request.

**Pull request merged**